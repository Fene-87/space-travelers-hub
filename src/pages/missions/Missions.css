table {
  width: 95%;
  margin: 2rem auto;
}

table,
th,
td {
  border: 1px solid #d8d8d7;
  border-collapse: collapse;
}

th,
td {
  padding: 12px 15px;
  text-align: left;
}

tbody tr:nth-child(odd) {
  background-color: #f2f2f2;
}

.mission_name {
  display: inline-block;
  border: none;
}

tbody .member-block,
.member-join {
  width: 10%;
  text-align: center;
}

tbody .member-block span {
  width: 100%;
  background-color: #6b7280;
  color: #fff;
  padding: 5px 10px;
  border-radius: 10px;
  font-size: 14px;
  border: none;
}

tbody .member-block .active {
  background-color: #22d3ee;
}

tbody .member-join button {
  border: 1px solid #525252;
  color: #525252;
  padding: 10px;
  border-radius: 5px;
  font-size: 14px;
  cursor: pointer;
}

tbody .member-join .outline-danger {
  border: 2px solid red;
  color: red;
}
