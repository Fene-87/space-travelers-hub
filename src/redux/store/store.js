import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from '../rockets/rocketsSlice';
import missionReducer from '../missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    missions: missionReducer,
  },
});

export default store;
