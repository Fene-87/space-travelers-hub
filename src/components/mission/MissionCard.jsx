import React from 'react';
import { PropTypes } from 'prop-types';
import { useDispatch } from 'react-redux';
// import Badge from 'react-bootstrap/Badge';
// import { joinAndLeaveMission } from '../../redux/missions/missionsSlice';
// import { Badge } from 'react-bootstrap';
import { joinMission, leaveMission } from '../../redux/missions/missionsSlice';

function MissionCard(props) {
  const {
    id, missionName, description, reserved,
  } = props;
  const dispatch = useDispatch();
  return (
    <tr>
      <th className="mission_name">{missionName}</th>
      <td className="description">
        {description}
      </td>
      <td className="member-block">
        <span
          className={reserved ? 'active' : 'not-active'}
        >
          {reserved ? 'Active Member' : 'Not A MEMBER'}
        </span>
      </td>
      <td className="member-join">
        <button
          type="button"
          className={reserved ? 'outline-danger' : 'outline-secondary'}
          onClick={reserved ? () => { dispatch(leaveMission(id)); }
            : () => { dispatch(joinMission(id)); }}
        //   onClick={() => dispatch(joinAndLeaveMission(id))}
        >
          {reserved ? 'Leave Mission' : 'Join Mission'}
        </button>
      </td>
    </tr>
  );
}

MissionCard.propTypes = {
  id: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default MissionCard;
