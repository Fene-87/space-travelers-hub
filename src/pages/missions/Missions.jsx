import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  selectAllMissions, getMissionsStatus, getMissionserror, fetchMissions,
} from '../../redux/missions/missionsSlice';
import Navbar from '../../components/navbar/Navbar';
import MissionCard from '../../components/mission/MissionCard';
import './Missions.css';

const Missions = () => {
  const dispatch = useDispatch();

  const missions = useSelector(selectAllMissions);
  const missionStatus = useSelector(getMissionsStatus);
  const missionError = useSelector(getMissionserror);

  useEffect(() => {
    if (missionStatus === 'idle') {
      dispatch(fetchMissions());
    }
  }, [missionStatus, dispatch]);

  let content;
  if (missionStatus === 'loading') {
    content = (
      <tr className="loader">
        <td>
          <p>
            Loading...
          </p>
        </td>
      </tr>
    );
  } else if (missionStatus === 'succeeded') {
    content = missions.map((mission) => (
      <MissionCard
        key={mission.mission_id}
        missionName={mission.mission_name}
        description={mission.description}
      />
    ));
  } else if (missionStatus === 'failed') {
    content = <p>{missionError}</p>;
  }

  return (
    <div>
      <Navbar />
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {content}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
