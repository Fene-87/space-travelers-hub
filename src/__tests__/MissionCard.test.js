import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store/store';
import Missions from '../pages/missions/Missions';
import MissionCard from '../components/mission/MissionCard';

describe('testing snapshot with Missions', () => {
  it('testing Mission component', () => {
    const mission = render(
      <Provider store={store}>
        <BrowserRouter>
          <Missions />
        </BrowserRouter>
      </Provider>,
    );
    expect(mission).toMatchSnapshot();
  });
  it('testing MissionCard component', () => {
    const missionCard = render(
      <Provider store={store}>
        <MissionCard />
      </Provider>,
    );
    expect(missionCard).toMatchSnapshot();
  });
});
