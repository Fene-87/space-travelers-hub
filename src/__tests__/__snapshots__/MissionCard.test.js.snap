// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshot with Missions testing Mission component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="navbar"
        >
          <div
            class="left-side"
          >
            <img
              alt="logo"
              class="logo-img"
              src="planet.png"
            />
            <h2
              class="nav-header"
            >
              Space Travelers' Hub
            </h2>
          </div>
          <div
            class="right-side"
          >
            <ul
              class="nav-links"
            >
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  Rockets
                </a>
              </li>
              <li
                class="boundary"
              >
                <a
                  class=""
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/profile"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </div>
        </section>
        <table>
          <thead>
            <tr>
              <th>
                Mission
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
              <th>
                 
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="loader"
            >
              <td>
                <p>
                  Loading...
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        class="navbar"
      >
        <div
          class="left-side"
        >
          <img
            alt="logo"
            class="logo-img"
            src="planet.png"
          />
          <h2
            class="nav-header"
          >
            Space Travelers' Hub
          </h2>
        </div>
        <div
          class="right-side"
        >
          <ul
            class="nav-links"
          >
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li
              class="boundary"
            >
              <a
                class=""
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class=""
                href="/profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </section>
      <table>
        <thead>
          <tr>
            <th>
              Mission
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
            <th>
               
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="loader"
          >
            <td>
              <p>
                Loading...
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshot with Missions testing MissionCard component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr>
        <th
          class="mission_name"
        />
        <td
          class="description"
        />
        <td
          class="member-block"
        >
          <span
            class="not-active"
          >
            Not A MEMBER
          </span>
        </td>
        <td
          class="member-join"
        >
          <button
            class="outline-secondary"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr>
      <th
        class="mission_name"
      />
      <td
        class="description"
      />
      <td
        class="member-block"
      >
        <span
          class="not-active"
        >
          Not A MEMBER
        </span>
      </td>
      <td
        class="member-join"
      >
        <button
          class="outline-secondary"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
